Expensify/react-native-live-markdown
# Improve error handling in `parseExpensiMarkToRanges`

### Details

This PR changes how we handle errors in `parseExpensiMarkToRanges`. Examples should now ignore styling if they encounter error while parsing instead of crashing. 

It also improves the error messages itself so that they explain what is failing, where and why 

### Related Issues

GH_LINK

### Manual Tests

Add a `new Error()` somewhere in the file or just make sure one of the existing errors gets thrown

### Linked PRs
